lib/conjur/graph.rb

```diff
@@ -144,7 +144,11 @@ def vertices
 
     def node_id_for role
       role = role.id if role.respond_to?(:id)
-      @node_ids[role]
+      node_id = @node_ids[role]
+      if node_id.nil?
+        node_id = @node_ids[role] = next_node_id
+      end
+      node_id
```

Comment:
It's not more efficient, but the class couldn't be Marshalled when it used a hash with a default function.